eclipse-che/che-operator
# Make workspace routes more user-friendly

Fixes https://github.com/eclipse/che/issues/20598

Depends on https://github.com/eclipse-che/che-server/pull/508

### What does this PR do?

#### Workspace URL
Before this PR, the workspace url would be something like:
```
https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/workspace13c0f27b694b4943/universal-developer-image/3100/
```

After this PR, the workspace URL becomes:
```
https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/<username>/<workspace name>/3100/
```

#### Public endpoints defined in devfile
For public endpoints defined in the devfile ([example](https://github.com/che-incubator/quarkus-api-example/blob/992ad93649a36ef6845592c69a2650199d5a3a5d/devfile.yaml#L18-L22))

Before this PR:
```
http://workspace294df8373ccb4bda-4.apps.che-dev.x6e0.p1.openshiftapps.com/food
```

After this PR:
```
http://<username>.<workspace name>.<endpoint name>.apps.che-dev.x6e0.p1.openshiftapps.com/food
```

### Screenshot/screencast of this PR


Workspace URL:
<img width="1212" alt="image" src="https://user-images.githubusercontent.com/83611742/236334786-cc9748c3-d945-4298-9bba-49d88f5920e0.png">

Route `spec.host`:
<img width="885" alt="image" src="https://user-images.githubusercontent.com/83611742/236334904-3940af95-4d86-403b-909f-77cbd59bbbfb.png">


### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20598

### How to test this PR?


```bash

chectl server:deploy \
     --installer operator \
     --platform openshift \
     --che-operator-image quay.io/dkwon17/che-operator:friendlypath 
```

~~1. After deploying Che, update the `che-server` image:~~ (CheCluster patch not needed anymore)
```
spec:
  components:
    cheServer:
      deployment:
        containers:
          - image: 'quay.io/dkwon17/che-server:user-profile-che-operator'
            name: che
```

2. Create and start a workspace with this project: https://github.com/che-incubator/quarkus-api-example
3. After workspace start, verify that the workspace IDE url is: `{CHE-HOST}/{USERNAME}/{DW-NAME}/3100/`
4. Confirm that there is an OpenShift route named `{DW-ID}-tools-8080-list-all-food`, where the `spec.host` is `{USERNAME}.{DW-NAME}.list-all-food.{INGRESS-DOMAIN}`


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
